freyja/sample_deconv.py

```diff
@@ -25,41 +25,32 @@ def buildLineageMap():
 	return mapDict
 
 
-def build_mix_and_depth_arrays(dirName,depthDir,muts):
-	df_mix = pd.DataFrame()
-	df_depths = pd.DataFrame()
-	fnames = os.listdir(dirName)
-
-	filetype =fnames[0].split('.')[-1]
-	for fn in tqdm(os.listdir(dirName)):
-		df = pd.read_csv(dirName+fn,sep='\t')
-		try:
-			df_depth  = pd.read_csv(depthDir+fn.split(filetype)[0]+'depth',sep='\t',header=None,index_col=1)
-		except:
-			continue
-		df['mutName'] = df['REF'] + df['POS'].astype(str) + df['ALT']### this only works for substitutions, but that's what we get from the usher tree
-		df = df.drop_duplicates(subset='mutName')
-		df.set_index('mutName',inplace=True)
-		keptInds = [dfi for jdi, dfi in  enumerate(df.index) if dfi  in muts]#meh... works but should be improved
-		df_mix = df_mix.append(pd.Series({kI:df.loc[kI,'ALT_FREQ'].astype(float) for kI in keptInds},name=fn))
-		df_depths = df_depths.append(pd.Series({kI:df_depth.loc[int(re.findall(r'\d+',kI)[0]),3].astype(float) for kI in muts},name=fn))
-	return df_mix,df_depths
-
-
-def reindex_dfs(df_barcodes,df_mix,df_depths):
+def build_mix_and_depth_arrays(fn, depthFn,muts):
+	df = pd.read_csv(fn,sep='\t')
+	df_depth  = pd.read_csv(depthFn,sep='\t',header=None,index_col=1)
+	df['mutName'] = df['REF'] + df['POS'].astype(str) + df['ALT']### this only works for substitutions, but that's what we get from the usher tree
+	df = df.drop_duplicates(subset='mutName')
+	df.set_index('mutName',inplace=True)
+	keptInds = [dfi for jdi, dfi in  enumerate(df.index) if dfi  in muts]#meh... works but should be improved
+	mix = pd.Series({kI:df.loc[kI,'ALT_FREQ'].astype(float) for kI in keptInds},name=fn)
```

Comment:
I think the following may be equivalent 
```suggestion
	mix = df.loc[keptInds, 'ALT_FREQ'].astype(float)
	mix.name = fn
```

Reply:
As a general (future) recommendation, it would be good to normalize types as early as possible -- like right after parse, done once and in bulk. It can be easy to forget to type cast later on, creating possibilities for difficult bugs, and doing it a lot is expensive.

Reply:
Agreed. And suggestion should be equivalent!